packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,350 @@
+/**
+ * External dependencies
+ */
+import { createElement, cloneElement, Fragment, isValidElement } from 'react';
+
+let indoc,
+	offset,
+	output,
+	stack;
+
+/**
+ * Matches tags in the localized string
+ *
+ * This is used for extracting the tag pattern groups for parsing the localized
+ * string and along with the map converting it to a react element.
+ *
+ * There are four references extracted using this tokenizer:
+ *
+ * match: Full match of the tag (i.e. <strong>, </strong>, <br/>)
+ * isClosing: The closing slash, it it exists.
+ * name: The name portion of the tag (strong, br) (if )
+ * isSelfClosed: The slash on a self closing tag, if it exists.
+ *
+ * @type {RegExp}
+ */
+const tokenizer = /<(\/)?(\w+)\s*(\/)?>/g;
+
+/**
+ * The filament describes and element and its children.
+ *
+ * This is used by the string iterator to track children attached to an element
+ * as the string is parsed. This allows for collecting nested elements in
+ * the string before creating the parent.
+ *
+ * @private
+ *
+ * @param {WPElement} element The element
+ *
+ * @return {Filament} A constructor for the Filament.
+ */
+function Filament( element ) {
+	return {
+		element,
+		children: [],
+	};
+}
```

Comment:
I get the idea of naming it as something entirely different, but `Filament` may be getting into the unnecessarily-convoluted domain.

This is used to track elements, what I've called components for the interpolate string, on the stack until we know we have parsed all of their children.

Considering that this code is simpler than what we use in the block parser we can honestly get rid of `Filament` entirely, I bet.

That is, we can probably add an `element` and `children` property to `Frame` and eliminate the second data structure. With blocks there were more properties to track and keep hold of and adding them into the `Frame` seemed like it was making `Frame` a bit unwieldy.

What if we did `Frame.element` and `Frame.children`?